docs/proposals/client-settings.md

```diff
@@ -0,0 +1,422 @@
+
+### Go
+
+```go
+package v1alpha1
+
+import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type ClientSettingsPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the ClientSettingsPolicy.
+    Spec ClientSettingsPolicySpec `json:"spec"`
+
+    // Status defines the state of the ClientSettingsPolicy.
+    Status ClientSettingsPolicyStatus `json:"status,omitempty"`
+}
+
+type ClientSettingsPolicySpec struct {
+    // TargetRef identifies an API object to apply the policy to.
+    // Object must be in the same namespace as the policy.
+    // Support: Gateway and HTTPRoute
+    TargetRef gatewayv1alpha2.PolicyTargetReference `json:"targetRef"`
+
+    // Default defines default policy configuration for the targeted resource.
+    // +optional
+    Default *ClientSettingsPolicyConfig `json:"default,omitempty"`
+}
+
+type ClientSettingsPolicyStatus struct {
+    // Conditions describe the current conditions of the ClientSettingsPolicy.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type ClientSettingsPolicyConfig struct {
+    // Body defines the client request body settings.
+    // +optional
+    Body *ClientBody `json:"body,omitempty"`
+
+    // KeepAlive defines the keep-alive settings.
+    // +optional
+    KeepAlive *ClientKeepAlive `json:"keepAlive,omitempty"`
+}
+
+// ClientBody contains the settings for the client request body.
+type ClientBody struct {
+    // MaxSize sets the maximum allowed size of the client request body.
+    // Setting size to 0 disables checking of client request body size.
+    // +optional
+    MaxSize *Size `json:"maxSize,omitempty"`
+
+    // Timeout defines a timeout for reading client request body. The timeout is set only for a period between
+    // two successive read operations, not for the transmission of the whole request body.
+    // +optional
+    Timeout *Duration `json:"timeout,omitempty"`
+}
+
+// ClientKeepAlive defines the keep-alive settings for clients.
+type ClientKeepAlive struct {
+    // Requests sets the maximum number of requests that can be served through one keep-alive connection.
+    // After the maximum number of requests are made, the connection is closed.
+    // +optional
+    Requests *int32 `json:"requests,omitempty"`
+
+    // Time defines the maximum time during which requests can be processed through one keep-alive connection.
+    // After this time is reached, the connection is closed following the subsequent request processing.
+    // +optional
+    Time *Duration `json:"time,omitempty"`
+
+    // Timeout defines the keep-alive timeouts for clients.
+    // +optional
+    Timeout *ClientKeepAliveTimeout `json:"timeout,omitempty"`
+}
+
+// ClientKeepAliveTimeout defines the timeouts related to keep-alive client connections.
+type ClientKeepAliveTimeout struct {
```

Comment:
Also changed this from `KeepAliveTimeout` to `ClientKeepAliveTimeout` for the same reason.